### PR TITLE
dxf: refactor error handling in task executor

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #57497

Problem Summary:

### What changed and how does it work?
- error of `GetStepExecutor` is fatal now
- we don't handle error of `Cleanup` now, only log it, previously we will mark a randomly subtask as failed, but if all subtasks are finished, this does nothing.
- the print stack logic in `onError` is removed now, I plan to add special initialized logger which will always print error stack to achieve the same effect.
- check context error using context itself, not by the error

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/disttask/framework/integrationtests/framework_test.go [hunk 1]
```diff
@@ -234,22 +234,6 @@ func TestGC(t *testing.T) {
 	}, 10*time.Second, 500*time.Millisecond)
 }
 
-func TestFrameworkSubtaskFinishedCancel(t *testing.T) {
```
Comment (D3Hunter) [inline pkg/disttask/framework/integrationtests/framework_test.go:237]: this case test what we do when a subtask is cancelled by other routines by `CancelRunningSubtask`, it's actually duplicate with `TestFrameworkRunSubtaskCancelOrFailed`, so merge them

File: pkg/disttask/framework/taskexecutor/task_executor.go [hunk 2]
```diff
@@ -236,58 +232,24 @@ func (e *BaseTaskExecutor) Run(resource *proto.StepResource) {
...
 }
 
 // RunStep start to fetch and run all subtasks for the step of task on the node.
 // return if there's no subtask to run.
-func (e *BaseTaskExecutor) RunStep(resource *proto.StepResource) (err error) {
-	defer func() {
-		if r := recover(); r != nil {
-			e.logger.Error("BaseTaskExecutor panicked", zap.Any("recover", r), zap.Stack("stack"))
-			err4Panic := errors.Errorf("%v", r)
-			err1 := e.updateSubtask(err4Panic)
```
Comment (D3Hunter) [inline pkg/disttask/framework/taskexecutor/task_executor.go:251]: should we fail the task on panic, or keep retry?

File: pkg/disttask/framework/taskexecutor/task_executor_test.go [hunk 3]
```diff
@@ -175,56 +173,51 @@ func TestTaskExecutorRun(t *testing.T) {
...
+		// same as previous case
+		taskExecutor.cancelRunStepWith(nil)
+		return grpcErr
+	})
 	mockStepExecutor.EXPECT().Cleanup(gomock.Any()).Return(nil)
 	err = taskExecutor.RunStep(nil)
 	require.EqualError(t, err, grpcErr.Error())
 	require.True(t, ctrl.Satisfied())
 
-	// 9. annotate grpc cancel
```
Comment (D3Hunter) [inline pkg/disttask/framework/taskexecutor/task_executor_test.go:210]: we check context cancel by using ctx.Err now, no need to check this one

**Review by wjhuang2016 — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]

**Pull request merged**